thorlcr/slave/slavmain.cpp

```diff
@@ -1214,21 +1173,60 @@ class CKJService : public CSimpleInterfaceOf<IKJService>, implements IThreaded,
     void addToKeyManagerCache(CKMContainer *kmc)
     {
         CriticalBlock b(kMCrit);
-        if (numCached == maxCachedKJManagers)
+        if (numKMCached == maxCachedKJManagers)
         {
             CKMContainer &oldest = cachedKMsMRU.item(0);
-            verifyex(removeKeyManager(oldest.queryHandle())); // also removes from cachedKMsMRU
+            auto it = cachedKMs.find(oldest.queryCtx().queryKey());
+            assertex(it != cachedKMs.end());
+            CKMKeyEntry *kme = it->second;
+            verifyex(kme->remove(kmc));
+            if (0 == kme->count())
+                cachedKMs.erase(it);
+            verifyex(cachedKMsMRU.zap(oldest));
```

Comment:
could use
    cachedKMsMRU.remove(0)
but not much more efficient since it matches the first element